exercises/src/test/scala/fpinscala/iomonad/IOSpec.scala

```diff
@@ -0,0 +1,42 @@
+package fpinscala.iomonad
+
+import org.specs2.mutable._
+import org.specs2.matcher.Scope
+
+class IOSpec extends Specification {
+
+  class IO3Fixture extends Scope {
+    import IO3._
+    import Console._
+
+    def countDown(from: Int): Free[Console, Unit] = {
```

Comment:
This is also a useful function for testing recursion complexity:
http://en.wikipedia.org/wiki/Ackermann_function#Table_of_values